### PR TITLE
Add AWS IAM support for DB connections.

### PR DESCRIPTION
**Pull request by vadimberezniker**
To support this functionality, users will need to use the new 'advanced' configuration stanzas. We need this to be able to generate the DSN using the short-lived tokens returned by AWS.

A typical use will look like this:

```
 advanced_data_source:
    driver: mysql
    use_aws_iam: true
    region: some-region
    endpoint: some-instance:3306
    username: some-user
    db_name: buildbuddy
```

Most users can continue to use the simpler 'data_source' option.

The other notable change is that we now control the creation of new DB connections so that we can generate a new DSN every time since the generated AWS IAM tokens are short-lived.

---

**Version bump**: Minor

### CODE REVIEW

File: server/util/db/db.go [hunk 1]
```diff
@@ -2,16 +2,19 @@ package db
...
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"database/sql"
+	"database/sql/driver"
 	"errors"
 	"flag"
 	"fmt"
+	golog "log"
```
Comment (tylerwilliams) [inline server/util/db/db.go:12]: can you move this down below

File: enterprise/server/cmd/server/BUILD [hunk 2]
```diff
@@ -87,6 +87,9 @@ go_binary(
 container_image(
     name = "base_image",
     base = "@buildbuddy_go_image_base//image",
+    files = [
+        "@aws_rds_certs//file",
+    ],
```
Comment (tylerwilliams) [inline enterprise/server/cmd/server/BUILD:92]: Can we move this to the //static target so it's also included in the binary bundle and will work for non-docker folks?

**Review by tylerwilliams — CHANGES_REQUESTED** [review]
This looks really good overall, just a couple of small nits

File: server/util/db/db.go (hunk 1 above)
Comment (vadimberezniker) [reply server/util/db/db.go]: Done.

**Review by tylerwilliams — APPROVED** [review]

**Pull request merged**